base-java/src/main/java/com/ibm/mq/samples/java/BasicGet.java

```diff
@@ -0,0 +1,133 @@
+package com.ibm.mq.samples.java;
+
+import com.ibm.mq.*;
+import com.ibm.mq.constants.MQConstants;
+import org.json.*;
+
+import java.io.FileReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Hashtable;
+
+public class BasicGet {
+
+    private static class MQDetails {
+        String QMGR;
+        String QUEUE_NAME;
+        String HOST;
+        String PORT;
+        String CHANNEL;
+        String USER;
+        String PASSWORD;
+        String KEY_REPOSITORY;
+        String CIPHER;
+    }
+
+    private static JSONArray endpoints;
+
+    public static void main(String[] args) {
+        loadEnv("env.json");
+
+        for (int i = 0; i < endpoints.length(); i++) {
+            System.out.println("Processing endpoint " + i);
+            JSONObject point = endpoints.getJSONObject(i);
+            MQDetails details = buildMQDetails(point);
+            processEndpoint(details);
+        }
+
+        System.out.println("Sample MQ GET application ending");
+    }
+
+    private static void loadEnv(String path) {
+        try {
+            String content = new String(Files.readAllBytes(Paths.get(path)));
+            JSONObject env = new JSONObject(content);
+            endpoints = env.getJSONArray("MQ_ENDPOINTS");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static MQDetails buildMQDetails(JSONObject endpoint) {
+        MQDetails mq = new MQDetails();
+        mq.QMGR = endpoint.getString("QMGR");
+        mq.QUEUE_NAME = endpoint.getString("QUEUE_NAME");
+        mq.HOST = endpoint.getString("HOST");
+        mq.PORT = String.valueOf(endpoint.getInt("PORT"));
+        mq.CHANNEL = endpoint.getString("CHANNEL");
+        mq.USER = endpoint.getString("APP_USER");
+        mq.PASSWORD = endpoint.getString("APP_PASSWORD");
+        mq.KEY_REPOSITORY = endpoint.optString("KEY_REPOSITORY", "");
+        mq.CIPHER = endpoint.optString("CIPHER", "");
+        return mq;
+    }
+
+    private static void processEndpoint(MQDetails details) {
+        MQQueueManager qMgr = null;
+        MQQueue queue = null;
+
+        try {
+            Hashtable<String, Object> props = new Hashtable<>();
+            String ccdtUrl = System.getenv("MQCCDTURL");
```

Comment:
Common shared check - should be a public method in the base class which all subclasses can invoke.